From 74ad44ba905ba8692815b0939d5364d37798d41e Mon Sep 17 00:00:00 2001
From: Mike FABIAN <mfabian@redhat.com>
Date: Thu, 14 Nov 2013 12:42:44 +0100
Subject: [PATCH] Commit candidate clicked on with the mouse

See https://bugzilla.redhat.com/show_bug.cgi?id=1029822
(Bug 1029822 - mouse selection and pagination numbering issue for ibus-typing-booster)
---
 ibus-typing-booster/engine/hunspell_table.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ibus-typing-booster/engine/hunspell_table.py b/ibus-typing-booster/engine/hunspell_table.py
index ec2b5cd..d02fc36 100644
--- a/ibus-typing-booster/engine/hunspell_table.py
+++ b/ibus-typing-booster/engine/hunspell_table.py
@@ -712,6 +712,11 @@ class tabengine (IBus.Engine):
         if len(tokens) > 1:
             self._editor._pp_phrase = tokens[-2]
 
+    def do_candidate_clicked(self, index, button, state):
+        phrase = self._editor.get_string_from_lookup_table_current_page(index)
+        if phrase:
+            self.commit_string(phrase + u' ')
+
     def do_process_key_event(self, keyval, keycode, state):
         '''Process Key Events
         Key Events include Key Press and Key Release,
-- 
1.8.4.2

